src/error.c

```diff
@@ -22,7 +22,7 @@ void jsonp_error_set_source(json_error_t *error, const char *source) {
 
     length = strlen(source);
     if (length < JSON_ERROR_SOURCE_LENGTH)
-        strncpy(error->source, source, length + 1);
+        strncpy(error->source, source, JSON_ERROR_SOURCE_LENGTH);
```

Comment:
Passing `JSON_ERROR_SOURCE_LENGTH` here seems unnecessary as the length has just been checked. We already know the length of `source` and have checked it will fit in the output buffer, therefore I think this is better:

```
         memcpy(error->source, source, length + 1);
```